Makefile

```diff
@@ -67,14 +67,14 @@ test:
 	g++ -std=c++0x -isystem ${GTEST_DIR}/include -I${GTEST_DIR} -pthread -c ${GTEST_DIR}/src/gtest-all.cc
 	ar -rv libgtest.a gtest-all.o
 	# Build tests
-	g++ -isystem ${GTEST_DIR}/include ${SESSION_TEST} ${GTEST_DIR}/src/gtest_main.cc libgtest.a $(OBJECT_FILES) -o session_test ${LDFLAGS} ${CXXFLAGS}
-	g++ -isystem ${GTEST_DIR}/include ${UTILS_TEST} ${GTEST_DIR}/src/gtest_main.cc libgtest.a $(OBJECT_FILES) -o utils_test ${LDFLAGS} ${CXXFLAGS}
-	g++ -isystem ${GTEST_DIR}/include ${PARSER_TEST} ${GTEST_DIR}/src/gtest_main.cc libgtest.a $(OBJECT_FILES) -o config_parser_test ${LDFLAGS} ${CXXFLAGS}
-	g++ -isystem ${GTEST_DIR}/include ${SERVER_TEST} ${GTEST_DIR}/src/gtest_main.cc libgtest.a $(OBJECT_FILES) -o server_test ${LDFLAGS} ${CXXFLAGS}
-	g++ -isystem ${GTEST_DIR}/include ${HANDLER_TEST} ${GTEST_DIR}/src/gtest_main.cc libgtest.a $(OBJECT_FILES) -o handler_test ${LDFLAGS} ${CXXFLAGS}
-	g++ -isystem ${GTEST_DIR}/include ${REQUEST_TEST}   ${GTEST_DIR}/src/gtest_main.cc libgtest.a ${OBJECT_FILES} -o  http_request_test ${LDFLAGS} ${CXXFLAGS}
-	g++ -isystem ${GTEST_DIR}/include ${RESPONSE_TEST}    ${GTEST_DIR}/src/gtest_main.cc libgtest.a ${OBJECT_FILES} -o  http_response_test ${LDFLAGS} ${CXXFLAGS}
-	g++ -isystem ${GTEST_DIR}/include ${RESPONSE_PARSER_TEST} ${GTEST_DIR}/src/gtest_main.cc libgtest.a ${OBJECT_FILES} -o response_parser_test ${LDFLAGS} ${CXXFLAGS}
+	g++ -isystem ${GTEST_DIR}/include ${SESSION_TEST} ${GTEST_DIR}/src/gtest_main.cc libgtest.a $(OBJECT_FILES) -o session_test ${LDFLAGS} -lboost_regex -lboost_iostreams ${CXXFLAGS}
```

Comment:
if you add -lboost_regex and -lboost_iostreams to LDFLAGS variable it'll fix this so you don't need to repeat it everywhere